tests/ChunkedEncoding.php

```diff
@@ -15,6 +15,10 @@ public static function chunkedProvider() {
 				"02\r\nab\r\n04\r\nra\nc\r\n06\r\nadabra\r\n0c\r\n\nall we got\n",
 				"abra\ncadabra\nall we got\n"
 			),
+			array(
+				"02;foo=bar;hello=world\r\nab\r\n04;foo=baz\r\nra\nc\r\n06;justfoo\r\nadabra\r\n0c\r\n\nall we got\n",
```

Comment:
RFC2616 states:

 > chunk-ext-name = token
 > chunk-ext-val  = token | quoted-string

Token cannot contain any white space, or any of the following: `(}|<>@,;:\"/[]?={}`. Quoted string can contain white space. 

What we have at the moment is still a relatively low-bar for what is interpreted of correctly encoded. I'm wary about making it _too_ strict as a misunderstanding here, when it's too strict, is more likely to lead to issues in production.

Happy to add the above restrictions with some additional tests